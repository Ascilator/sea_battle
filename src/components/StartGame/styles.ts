import styled from 'styled-components';
import { StylesType } from '@/types';

export const StyledMainScreenWrapper = styled.div<StylesType>`
  height: 100%;
  max-width: 1240px;
  margin: 0 auto;
  padding: 0 20px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
