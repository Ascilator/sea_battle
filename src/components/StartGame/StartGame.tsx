import { FC } from "react";
import { useHistory } from "react-router";
import { v4 as uuidv4 } from "uuid";
import { Button } from "@/controls";

import { StyledMainScreenWrapper } from "./styles";

const StartGame: FC = () => {
  const history = useHistory();

  const generateRoom = () => {
    history.push(uuidv4());
  };
  return (
    <StyledMainScreenWrapper>
      <Button text={"Start Game"} color={"#000080"} onClick={generateRoom} />
    </StyledMainScreenWrapper>
  );
};

export { StartGame };
