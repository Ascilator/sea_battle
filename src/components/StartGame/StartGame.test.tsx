import renderer from 'react-test-renderer';
import { StartGame } from './StartGame';

test('StartGame render', () => {
  const startGame = renderer.create(<StartGame />).toJSON();
  expect(startGame).toMatchSnapshot();
});
