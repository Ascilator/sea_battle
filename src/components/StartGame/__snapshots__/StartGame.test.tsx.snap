// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGame render 1`] = `
<div
  className="sc-gsDKAQ isJwyh"
>
  <button
    className="sc-bdvvtL fsPlRH"
    color="#000080"
    onClick={[Function]}
  >
    <span>
      Start Game
    </span>
  </button>
</div>
`;
